bite-size-notes/week-6-b-free-final-time-PMP.md

```diff
@@ -0,0 +1,231 @@
+
+## Scope
+
+These notes will focus on introducing free final time on a simple linear system with theory and code, explain difficulties with implementation on nonlinear and higher-order linear systems, and provide resources to explore the subject further.
+
+Learning objectives include...
+* Discovering the difference between functions and functionals.
+* Understanding the similarities of extremizing functions of a functional with finding critical points.
+* Determining necessary conditions for optimality with the Hamiltonian under Pontryagin's Minimum Principle.
+* Constructing a free final time problem of a simple linear system.
+
+The reader is expected to have prior knowledge of undergraduate-level controls and topics such as basic optimization and the Lagrangian.
+
+## Prerequisites
+
+
+### Nomenclature
+<details close>
+<summary> <em> (open to view)</em> </summary>
+<br>
+<div markdown="1">
+
+| Term            | Definition                                      |
+|-----------------|-------------------------------------------------|
+| $a$ | acceleration |
+| $H$ | Hamiltonian |
+| $J$ | cost function/functional |
+| $L$ | Lagrangian |
+| $t$ | time |
+| $T$ | terminal time |
+| $u$ | control |
+| $v$ | velocity |
+| $x$ | position in $x$-direction |
+| $x_g$ | goal position in $x$-direction |
+| $y$ | position in $y$-direction |
+| $\theta$ | angle |
+| $\lambda$ | co-state |
+| $\omega$ | angular velocity |
+| $\Psi$ | terminal cost |
+
+</div>
+</details>
+
+### Motivation
+Why do we need new math to solve free final time?
+
+For free final time problem, terminal time $T$ is no longer fixed. Thus when finding an optimal solution, the state $z$ and control $u$ are no longer the optimization variables in a cost function, but rather $z(t)$, $u(t)$ are optimization *functions* and $T$ is an optimization variable in a cost *functional*.
+
+Put another way: rather than finding extrema (minima/maxima) values for a function, we instead find extrema functions, functions that minimize or maximize a functional.
+
+### Functionals
+Functionals are mappings from a set of functions to the set of real numbers. Intuitively, one can think of them as "functions of functions", as functions are mappings from a set of numbers to another set of numbers. The calculus of variations uses small changes of functions and functionals, known as variations, to determine the extrema of functionals.
+
+The purpose behind using the calculus of variations for free final time is to find a extrema function that minimizes/maximizes a functional: analogous to finding the critical points by setting the derivative of a function equal to zero.
+
+Our optimization cost is then represented by a cost functional $J[x(t),u(t),t]$ rather than a cost function $J\big(x(t),u(t),t\big)$. More precisely,
+
+$$\underbrace{J[z(t),u(t),t]}_{\text{cost functional}} = \int_0^T \underbrace{L\big(z(t), u(t), t\big)}_{\text{cost-to-go}} \,dt +  \underbrace{\Psi(T)}_{\text{terminal cost}}$$
```

Comment:
Notation clash: either $x$ or $z$ for the generic state

Reply:
Thanks for the catch!